File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-java/src/main/java/com/example/statemachine/simple/doument/config/action/SendEmailAction.java
package com.example.statemachine.simple.doument.config.action;

import com.example.statemachine.simple.doument.DocumentEvent;
import com.example.statemachine.simple.doument.DocumentState;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.statemachine.StateContext;
import org.springframework.statemachine.action.Action;

public class SendEmailAction implements Action<DocumentState, DocumentEvent> {

  private static final Logger logger
    = LoggerFactory.getLogger(SendEmailAction.class);

  @Override
  public void execute(StateContext<DocumentState, DocumentEvent> context) {
    logger.info("이메일 전송 : [문서가 대외 공개되었습니다.]");
  }
}

File: simple-statemachine-with-persistence/using-statemachine-recipe/simple-statemachine-kotlin/src/main/kotlin/com/example/statemachine/simple/document/config/action/SendEmailAction.kt
package com.example.statemachine.simple.document.config.action

import com.example.statemachine.simple.document.DocumentEvent
import com.example.statemachine.simple.document.DocumentState
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.statemachine.StateContext
import org.springframework.statemachine.action.Action

class SendEmailAction : Action<DocumentState, DocumentEvent> {

  private val logger: Logger = LoggerFactory.getLogger(SendEmailAction::class.java)

  override fun execute(p0: StateContext<DocumentState, DocumentEvent>) {
    logger.info("이메일 전송 : [문서가 대외 공개되었습니다.]")
  }
}
